modules/apps/document-library/document-library-analytics/src/main/resources/META-INF/resources/dynamic_include/top_head.jsp

```diff
@@ -64,14 +64,17 @@
 		}
 	}
 
-	document.addEventListener('DOMContentLoaded', function () {
+	var onPortletReady = function() {
 		document.body.addEventListener('click', handleDownloadClick);
-	});
+	};
 
 	var onDestroyPortlet = function () {
 		document.body.removeEventListener('click', handleDownloadClick);
+
+		Liferay.detach('portletReady', onPortletReady);
 		Liferay.detach('destroyPortlet', onDestroyPortlet);
 	};
 
+	Liferay.on('portletReady', onPortletReady);
```

Comment:
There is another method called `Liferay.once` what do you think in use instead of `on` with `detach`?

Reply:
Done!